### ISSUE TITLE
Crash of simulator during CN.addToQueueAndWait(request)

### ISSUE THREAD

**Issue by timovandeput** [codenameone/CodenameOne#2655]
The simulator fatally crashes after submitting a REST request on line 513 of RequestBuilder.java

I am running the Simulator on my MacBook Pro from Intellij.

Version.properties lists:
>    #Sat Dec 29 17:35:24 CET 2018
>    CodenameOne_SRCzip=51
>    CodenameOneJar=51
>    CodeNameOneBuildClientJar=14
>    CLDC11Jar=17
>    JavaSEJar=51

The crash reports:
>    A fatal error has been detected by the Java Runtime Environment:
>   
>    SIGSEGV (0xb) at pc=0x0000000110decd8a, pid=1740, tid=31751
>    
>    JRE version: OpenJDK Runtime Environment (11.0.1+13) (build 11.0.1+13)
>    Java VM: OpenJDK 64-Bit Server VM (11.0.1+13, mixed mode, tiered, compressed oops, g1 gc, bsd-amd64)
>    Problematic frame:
>    V  [libjvm.dylib+0x3ecd8a]  jni_CallStaticVoidMethodV+0xdf

Please help.

**Comment by codenameone** [comment]
A segfault generally means a JDK bug which makes sense for JDK 11. I suggest checking if there's a JDK update available.

**Comment by timovandeput** [comment]
I downgraded to JDK 10.0.2, and now it works; thanks!

But at startup of the simulator, it does mention:
> InteropFactory: cannot load com.sun.javafx.embed.swing.newimpl.InteropFactoryN

(Not sure if it has anything to do with the reported crash, but JDK 11 has gotten very strict at access outside modules.)

Note that JDK 11.0.1 is current "latest-greatest" on Mac, and more people might be impacted by this compatibility issue.

Thanks for the quick response; happy New Year!

**Comment by timovandeput** [comment]
Did anybody try Java 12 yet?

**Comment by codenameone** [comment]
No. JDK 11.x would be a better target here. JDK 11 is an LTS release and we only focus on those as it's too much of a rat race to keep up with constant compatibility breaking JDK updates.
